5.0 Draft/IDD/IDDs Orchestration System/eu.arrowhead.orchestration-pull-http-json.yml

```diff
@@ -0,0 +1,303 @@
+        the service definition and at least one interface must be specified. 
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ServiceRequestForm'
+      responses:
+        '200':
+          description: Successful orchestration response.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/OrchestrationResponse'
+
+  /orchestration/orchestration-pull/{id}:
+    get:
+      summary: Get Orchestration by system id
+      parameters:
+        - name: id
+          in: path
+          required: true
+          description: ID of requesting system
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OrchestrationResponse'
+        '404':
+          description: System id not found
+          content:
+            application/json:
+              example:
+                error: Orchestration failed, system id not found. 
+
+
+components:
+  schemas:
+    Address:
+      description: |
+        Identifies a _transport_ and a _location_.
+
+        The _transport_ identifies the base protocol that facilitates addressing a specific service
+        instance. Examples of transports that can be supported are `tcp4`, `tcp6`, `udp4`, `udp6`
+        and `unix`. A protocol only counts as a transport if it both (A) provides a way of
+        addressing and, by extension, sending messages to service providers and consumers, as well
+        as (B) does not build upon another protocol also providing this capability. I other words,
+        TLS and DTLS are _not_ transports, because they build upon the TCP and UDP protocols, which
+        satisfy condition A.
+
+        What the _location_ part consists of depends on what transport is identified. If the
+        transport is `tcp4` or `udp4`, the location is an IPv4 address expressed as four decimal
+        numbers separated by dots, a colon and a port number, such as in `192.168.3.22:64075`. If
+        the transport is `tcp6` or `udp6`, the location is an IPv6 address, rendered as described in
+        [RFC 5952](https://www.rfc-editor.org/rfc/rfc5952), within square brackets, followed by a
+        colon and a port number. If the transport is `unix`, the location is an absolute filesystem
+        path to a Unix socket file.
+      type: string
+      pattern: ^(?<transport>[^:]+):\w*(?<location>.*)$
+      example: tcp4:192.168.0.7:45326
+
+    Auth-tokens:
+      description: |
+        An (k,v) object containing authorization tokens for each interface contained in the response
+      type: object
+      example: {"interfaceName1":"token1"}
+
+    interface:
```

Comment:
There are plenty of problems with these interface triplets, which were used in Arrowhead 4.0. Firstly, not all transports allow you to choose your own encoding for payloads. JSONRPC requires you to use JSON, for example. Also, some protocols do not exist in secure form or insecure form.

I proposed a service registration request message some time ago that replaces the interface triplet with an actual interface object. You can look at it here: [[1]](https://github.com/eclipse-arrowhead/roadmap/issues/22#issuecomment-992862174) [[2]](https://github.com/eclipse-arrowhead/roadmap/issues/22#issuecomment-1002057239).